### PR TITLE
feat(rxApp): hide navigation if embedded

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: https://jira.rax.io/browse/FRMW-796

### LGTMs
- [x] Dev LGTM
- [x] Dev LGTM
- [x] Design LGTM
- [x] QE LGTM

### Description

* Hide nav when app detects it is in iframe
* correct styling for global notifications without nav
* correct external "Source" link when embedded
* add embedded rxApp example

### CODE REVIEW

**Comment by CITguy** [comment]
### Metadata (w/detect support)

*business as usual*

metadata-with-detection-ok

### Metadata (embedded w/ detect support)

*navigation is hidden in iframe*

metadata-within-iframe-ok

**Comment by CITguy** [comment]
### rxApp (w/ embedded example)
rxapp-with-embedded-example

File: demo/templates/modules/showModule.html [hunk 1]
```diff
@@ -15,7 +15,12 @@
           <a href="ngdocs/index.html#/api/{{ module.category }}.{{ module.api }}/" class="module-api button">View API</a>
         </span>
       </span>
-      <a href="https://github.com/rackerlabs/encore-ui/tree/master/src/{{ module.category }}/{{ module.name }}/" class="module-source button">View Source</a>
+      <a
+        href="https://github.com/rackerlabs/encore-ui/tree/master/src/{{ module.category }}/{{ module.name }}/"
+        class="module-source button"
+        target="_parent">
```
Comment (CITguy) [inline demo/templates/modules/showModule.html]: Without this, the button won't work in an iframe.

File: src/components/rxApp/rxApp.less [hunk 2]
```diff
@@ -528,14 +536,11 @@
...
-      z-index: 20;  // accommodate rxSelect .fake-select class
+      z-index: 20; // accommodate rxSelect .fake-select class
       top: 0;
       left: 300px;
-      right: 100px;
+      right: 0px;
       opacity: 0.95;
-      .rx-notification {
-        width: 100%;
-      }
```
Comment (CITguy) [inline src/components/rxApp/rxApp.less]: `.rx-notification` is a div (block element) that natively expands to fill width of parent. This style is unnecessary.

File: src/components/rxApp/templates/rxApp.html [hunk 3]
```diff
@@ -3,8 +3,8 @@
         <span class="notification-text">{{ warningMessage }}</span>
     </div>
 </div>
-<div class="rx-app" ng-class="{collapsible: collapsibleNav === 'true', collapsed: collapsedNav, 'warning-bar': isWarning, preprod: isPreProd}" ng-cloak>
-    <nav class="rx-app-menu">
+<div class="rx-app" ng-class="{collapsible: collapsibleNav === 'true', collapsed: collapsedNav, 'warning-bar': isWarning, preprod: isPreProd, 'embedded': isEmbedded}" ng-cloak>
+    <nav class="rx-app-menu" ng-show="!isEmbedded">
```
Comment (CITguy) [inline src/components/rxApp/templates/rxApp.html]: Must use `ng-show` or else it becomes very challenging to correct/update specs (because specs run in an iframe, apparently).

File: src/components/rxApp/docs/rxApp.html [hunk 4]
```diff
@@ -1,21 +1,32 @@
...
-        account-number="12345">
-            <p class="clear">Click a link in the menu to see the active state change</p>
-            <p>Click the toggle to hide the menu</p>
-            <button ng-click="changeSubtitle()" class="changeSubtitle button">Change Subtitle</button>
-        </rx-page>
-    </rx-app>
+  <h3>Customized rxApp (collapsible)</h3>
+  <rx-app collapsible-nav="true" site-title="My App" id="custom-rxApp" menu="customMenu" new-instance="true" hide-feedback="true">
+    <rx-page
+      unsafe-html-title="'Customized Page <a href=&quot;http://rackspace.com&quot;>Title</a>'"
```
Comment (halkon) [inline src/components/rxApp/docs/rxApp.html]: OoooooooOOoOOOoOOoOOo :smiling_imp:

File: src/components/rxApp/scripts/rxApp.js [hunk 5]
```diff
@@ -44,6 +44,8 @@ angular.module('encore.ui.rxApp')
 
             scope.isWarning = scope.isPreProd || scope.isLocalNav;
 
+            scope.isEmbedded = $window.self !== $window.top;
```
Comment (halkon) [inline src/components/rxApp/scripts/rxApp.js]: Sneaky sneaky

**Comment by halkon** [comment]
LGTM

**Comment by parlarjb** [comment]
LGTM

**Comment by parlarjb** [comment]
Does the iframe work properly when changing `ng-src`? i.e. start with one URL, have it load, then change to a different one?

**Comment by CITguy** [comment]
@glynnis: the blue vertical bar is not visible on an embedded app. See screenshot in [comment above](https://github.com/rackerlabs/encore-ui/pull/1615#issuecomment-205497865). In Origin, they'd see (1) the Origin nav, and (2) the main content area of the embedded app.

File: src/components/rxApp/templates/rxApp.html (hunk 3 above)
Comment (Droogans) [inline src/components/rxApp/templates/rxApp.html]: May be an underlying decision made by PhantomJS? Interesting.

File: demo/templates/modules/showModule.html (hunk 1 above)
Comment (Droogans) [inline demo/templates/modules/showModule.html]: I'm not getting this to work. I'm clicking the links inside of the iframe, but my browser's url is not changing. Is that expected?

Comment (CITguy) [inline demo/templates/modules/showModule.html]: TBH, this change is borderline scope creep, because we don't need to update all links in the documentation to support embedding. All this PR needs to do is hide the left-hand nav. I'm going to revert this change.

**Comment by CITguy** [comment]
@Droogans: I've removed that modification to the `showModule.html` template. Please review at earliest convenience.

**Comment by Droogans** [comment]
Much better now that I can click stuff!
